test/functorch/test_aotdispatch.py

```diff
@@ -952,7 +952,6 @@ def assert_compiler(gm: torch.fx.GraphModule, _):
     xfail('linalg.eig'),
     xfail('scatter_reduce', 'prod'),
 
-    # non-deterministic
```

Comment:
Why remove this?

Reply:
I thought the non-determinism was caused by the uninitialized memory 😛

Reply:
Isn't the non-determinism that there may be multiple values mapping to the same index in the result, and which value appears there is nondeterministic? cc @kurtamohler